crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
It seems to me that even if this were to change to fail per-SCT, if the caller attempts to set a source to an SCT that couldn't have come from that source, they've done something wrong and should simply be told "FAIL".

If the caller wants to know how many of their SCTs are usable, they can have a method that sanity-checks the stack. Otherwise, you're gonna end up with something like
1) set source - 3 of 5 succeeded
2) set signature - 3 of 5 succeeded
How many good SCTs do I have now?